src/http/sr-modules-validators/sr-modules-validators.service.ts

```diff
@@ -5,164 +5,141 @@ import {
-  }
-
-  private async getOperatorOldestValidators(
-    operatorId: number,
-    filters: ValidatorsQuery,
-  ): Promise<{ validators: Validator[]; meta: ConsensusMeta }> {
-    // get used keys for operator
-    const { keys, meta: elMeta } = await this.curatedService.getKeysWithMeta({
-      used: true,
-      operatorIndex: operatorId,
-    });
-
-    // check if elMeta is not null
-    // if it is null, it means keys db is empty and Updating Keys Job is not finished yet
-    if (!elMeta) {
-      this.logger.warn(`EL meta is empty, maybe first Updating Keys Job is not finished yet.`);
-      throw httpExceptionTooEarlyResp();
-    }
-
-    const pubkeys = keys.map((pubkey) => pubkey.key);
-    const percent =
-      filters?.max_amount == undefined && filters?.percent == undefined ? DEFAULT_EXIT_PERCENT : filters?.percent;
-
-    const result = await this.validatorsService.getOldestValidators({
-      pubkeys,
-      statuses: VALIDATORS_STATUSES_FOR_EXIT,
-      max_amount: filters?.max_amount,
-      percent: percent,
-    });
-
-    if (!result) {
-      // if result of this method is null it means Validators Registry is disabled
-      throw new InternalServerErrorException(VALIDATORS_REGISTRY_DISABLED_ERROR);
-    }
-
-    const { validators, meta: clMeta } = result;
-
-    // check if clMeta is not null
-    // if it is null, it means keys db is empty and Updating Validators Job is not finished yet
-    if (!clMeta) {
-      this.logger.warn(`CL meta is empty, maybe first Updating Validators Job is not finished yet.`);
-      throw httpExceptionTooEarlyResp();
-    }
-
-    // We need EL meta always be actual
-    if (elMeta.blockNumber < clMeta.blockNumber) {
-      this.logger.warn('Last Execution Layer block number in our database older than last Consensus Layer');
-      // add metric or alert on breaking el > cl condition
-      // TODO: what answer will be better here?
-      // TODO: describe in doc
-      throw new InternalServerErrorException(
-        'Last Execution Layer block number in our database older than last Consensus Layer',
-      );
-    }
-
-    return { validators, meta: clMeta };
+    return {
+      data,
+      meta: {
+        clBlockSnapshot: clBlockSnapshot,
+      },
+    };
   }
 
   private createExitValidatorList(validators: Validator[]): ExitValidator[] {
     return validators.map((v) => ({ validatorIndex: v.index, key: v.pubkey }));
   }
 
-  private createExitPresignMessageList(validators: Validator[], clMeta: ConsensusMeta): ExitPresignMessage[] {
+  private createExitPresignMessageList(validators: Validator[], clMeta: CLBlockSnapshot): ExitPresignMessage[] {
     return validators.map((v) => ({ validator_index: String(v.index), epoch: String(clMeta.epoch) }));
   }
 
-  private disabledRegistry() {
+  private isRegistryDisabled() {
     return !this.configService.get('VALIDATOR_REGISTRY_ENABLE');
   }
+
+  private async getOperatorOldestValidators(
```

Comment:
Why this is not just get validators method with filter by node operator, sorted by created time and limited with some amount?

Reply:
will consider here join in the iteration